### DISCUSSION TITLE
Alter type rename value of enum

### DISCUSSION THREAD

**Discussion by razvan286** [sqlalchemy/alembic#1206] [Usage Questions] [answered]
I have an enum named `ProjectStatus` and a db column: `status = db.Column(db.Enum(ProjectStatus))`.

Within a migration I am altering the type of `projectstatus` by renaming a value of it such as:  
`op.execute("ALTER TYPE projectstatus RENAME VALUE 'READY' TO 'COMPLETED'")`

Does the alter type need an explicit data migration for already existing data with status `'READY'` to` 'COMPLETED'`? Or is it done automatically?

**Comment by zzzeek** [comment]
right now Alembic doesn't have any functionality for taking care of PostgreSQL or MySQL ENUM datatypes.   It's a long term goal that's on the [autogenerate for enums](https://github.com/sqlalchemy/alembic/labels/autogenerate%20for%20enums) tag.    so for now you would need to use `op.execute("ALTER TYPE...")` as needed.

**Comment by CaselIT** [comment, marked as answer]
Hi,

> Does the alter type need an explicit data migration for already existing data with status 'READY' to 'COMPLETED'? Or is it done automatically?

Assuming postgresql, alter type is a command of the db, it's not something provided by alembic, so this is not really a database question.
You would refer to their doc for it https://www.postgresql.org/docs/current/sql-altertype.html.

To answer the question, no other change is needed to the existing data
